cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -578,7 +579,17 @@ struct IdentityHash {
     return combined;
   }
 
-  CUDA_HOST_DEVICE_CALLABLE result_type operator()(const Key& key) const
+  template <typename return_type = result_type>
+  CUDA_HOST_DEVICE_CALLABLE std::enable_if_t<!std::is_arithmetic<Key>::value, return_type>
+  operator()(const Key& key) const
+  {
+    release_assert(false && "IdentityHash does not support this data type");
+    return 0;
+  }
```

Comment:
Why is this necessary?  I think that this code turns a compile time error into a runtime error, which is not good for developers because it will cause them to find their coding errors later.

If you remove this, does the code still compile?  If so then this code is never generating anything so it can be removed.

Reply:
This function object is invoked via the `type_dispatcher` which will instantiate it for all possible libcudf types. We need to provide a valid instantiation for all types. This includes types that should never actually be invoked (as seen above).

Reply:
Okay, so if these lines of code are removed then cudf will no longer compile successfully?

Reply:
For this case yes. `static_cast` a string to integer should fail at compile time.